File: ABC189/C-MandarinOrange.py
#pythonでは時間内に解けず

N = int(input())
Alist = list(map(int,input().split()))

#lを固定し，rを増やながら，xを更新する
#O(N**2)で解ける
ans = 0
for l,now in enumerate(Alist):
  #lを固定
  x = now
  for r in range(l,N):
    #区間最小値を選ぶ
    x = min(x,Alist[r])
    #lからrまでの区間最小値の和を求め，値が大きければ更新
    ans = max(ans,x*((r-l)+1))
    
print (ans)

File: ABC189/C-MandarinOrange.cpp
#include<bits/stdc++.h>

using namespace std;
int a[10010];

//main関数
int main(){
	int n;
    //入力
	cin >> n;
    //リスト入力　
	for(int i=0;i<n;i++)cin >> a[i];
	int ans=0;
	for(int l=0;l<n;l++){
        //区間最小値を定義
		int x=a[l];
		for(int r=l;r<n;r++){
			x=min(x,a[r]);
			ans=max(ans,x*(r-l+1));
		}
	}
	cout << ans;
}